parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriteStoreV2.java

```diff
@@ -158,20 +159,25 @@ private void sizeCheck() {
       }
       long rowsToFillPage =
           usedMem == 0 ?
-              MAXIMUM_RECORD_COUNT_FOR_CHECK
+              parquetProperties.getMaxRowCountForPageSizeCheck()
               : (long)((float)rows) / usedMem * remainingMem;
       if (rowsToFillPage < minRecordToWait) {
         minRecordToWait = rowsToFillPage;
       }
     }
     if (minRecordToWait == Long.MAX_VALUE) {
-      minRecordToWait = MINIMUM_RECORD_COUNT_FOR_CHECK;
+      minRecordToWait = parquetProperties.getMinRowCountForPageSizeCheck();
+    }
+
+    if(parquetProperties.isEstimateNextSizeCheck()) {
```

Comment:
This looks good to me now.

Nit: the comments aren't needed and reference values no longer used. Maybe just change the "will check again halfway" by adding " if it is between min and max"